volmdlr/faces.py

```diff
@@ -5076,19 +5112,24 @@ def line_intersections(self,
 
         return intersections
 
-    def linesegment_intersections(self,
-                                  linesegment: vme.LineSegment3D,
-                                  ) -> List[volmdlr.Point3D]:
-        intersections = []
+    def linesegment_intersections(self, linesegment: vme.LineSegment3D) -> List[volmdlr.Point3D]:
+        linesegment_intersections = []
         for intersection in self.surface3d.linesegment_intersections(linesegment):
             if self.point_belongs(intersection):
-                intersections.append(intersection)
-        if not intersections:
+                linesegment_intersections.append(intersection)
+        if not linesegment_intersections:
             for prim in self.outer_contour3d.primitives:
-                intersection = prim.linesegment_intersection(linesegment)
-                if intersection is not None:
-                    if intersection not in intersections:
-                        intersections.append(intersection)
+                intersections = prim.linesegment_intersections(linesegment)
+                for intersection in intersections:
+                    if intersection not in linesegment_intersections:
+                        linesegment_intersections.append(intersection)
+        return linesegment_intersections
```

Comment:
In short, this method has been modified because the methods linesegment_intersections from classes in edges.py have been updated to return always a list of intersections and NO MORE `None` if there aren't any intersections.